clientv3/lease.go

```diff
@@ -98,12 +87,14 @@ type Lease interface {
 	// TimeToLive retrieves the lease information of the given lease ID.
 	TimeToLive(ctx context.Context, id LeaseID, opts ...LeaseOption) (*LeaseTimeToLiveResponse, error)
 
-	// KeepAlive keeps the given lease alive forever.
-	KeepAlive(ctx context.Context, id LeaseID) (<-chan *LeaseKeepAliveResponse, error)
+	// KeepAlive keeps the given lease alive forever. If the keepalive response posted to
+	// the channel is not consumed immediately, the lease client will continue sending keep alive requests
+	// to the etcd server at least every second until latest response is consumed.
+	KeepAlive(ctx context.Context, id LeaseID) LeaseKeepAliveChan
```

Comment:
how people handles the error anyway today? probably with a simple retry the same as what we do internally now, right? if we stop returning the error, nothing is going to change for them. they can throw away the error handling code at their pace.

but, yea, we have to communicate this change. it just means that peoples' code can still compile and probably still work as expected.